pkg/probe/http/http.go

```diff
@@ -22,20 +22,67 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/url"
+	"path/filepath"
 	"time"
 
 	utilnet "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/kubernetes/pkg/probe"
 	"k8s.io/kubernetes/pkg/version"
 
+	"crypto/x509"
 	"github.com/golang/glog"
 )
 
-func New() HTTPProber {
+const (
+	defaultKubeletClientCertificateFile = "kubelet-client.crt"
+	defaultKubeletClientKeyFile         = "kubelet-client.key"
+	defaultKubeletCACertificateFile     = "kubelet-ca.crt"
+)
+
+func NewHttpProber() HTTPProber {
 	tlsConfig := &tls.Config{InsecureSkipVerify: true}
 	return NewWithTLSConfig(tlsConfig)
 }
 
+func NewHTTPSProber(certDir string) HTTPProber {
+	if certDir == "" {
+		glog.Errorln("path to HTTPS prober key/cert is not set")
+		return NewHttpProber()
+	}
+
+	certPath, err := filepath.Abs(filepath.Join(certDir, defaultKubeletClientCertificateFile))
+	if err != nil {
+		glog.Errorf("unable to build HTTPS prober cert path: %v", err)
+		return NewHttpProber()
+	}
+	keyPath, err := filepath.Abs(filepath.Join(certDir, defaultKubeletClientKeyFile))
+	if err != nil {
+		glog.Errorf("unable to build HTTPS prober key path: %v", err)
+		return NewHttpProber()
+	}
+	cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		glog.Errorf("unable to read and parse key/cert for HTTPS prober: %v", err)
+		return NewHttpProber()
+	}
+
+	caCert, err := ioutil.ReadFile(filepath.Join(certDir, defaultKubeletCACertificateFile))
+	if err != nil {
+		glog.Errorf("unable to read CA cert: %v", err)
+		return NewHttpProber()
+	}
+	certPool := x509.NewCertPool()
+	certPool.AppendCertsFromPEM(caCert)
+
+	tlsConfig := &tls.Config{
+		Certificates:       []tls.Certificate{cert},
+		RootCAs:            certPool,
```

Comment:
I don't think you need to set `RootCAs` since `InsecureSkipVerify` is set to true

Reply:
That's right.

Reply:
Thanks for pointing that out. Removed RootCA